src/test/javascript/spec/integration/code-editor/code-editor-container.integration.spec.ts

```diff
@@ -302,46 +289,46 @@ describe('CodeEditorContainerIntegration', () => {
-        expect(container.editorState).to.equal(EditorState.CLEAN);
-        expect(container.buildOutput.isBuilding).to.be.false;
-        expect(container.unsavedFiles).to.be.empty;
+        expect(container.commitState).toBe(CommitState.COULD_NOT_BE_RETRIEVED);
+        expect(container.editorState).toBe(EditorState.CLEAN);
+        expect(container.buildOutput.isBuilding).toBe(false);
+        expect(container.unsavedFiles).toStrictEqual({});
 
         // file browser
-        expect(checkIfRepositoryIsCleanStub).to.have.been.calledOnce;
-        expect(getRepositoryContentStub).to.not.have.been.calledOnce;
-        expect(container.fileBrowser.errorFiles).to.deep.equal(extractedErrorFiles);
-        expect(container.fileBrowser.unsavedFiles).to.be.empty;
+        expect(checkIfRepositoryIsCleanStub).toHaveBeenCalledTimes(1);
+        expect(getRepositoryContentStub).not.toHaveBeenCalled();
+        expect(container.fileBrowser.errorFiles).toEqual(extractedErrorFiles);
+        expect(container.fileBrowser.unsavedFiles).toHaveLength(0);
 
         // ace editor
-        expect(container.aceEditor.isLoading).to.be.false;
-        expect(container.aceEditor.annotationsArray.map((a) => omit(a, 'hash'))).to.deep.equal(extractedBuildLogErrors);
-        expect(container.aceEditor.commitState).to.equal(CommitState.COULD_NOT_BE_RETRIEVED);
+        expect(container.aceEditor.isLoading).toBe(false);
+        expect(container.aceEditor.annotationsArray.map((a) => omit(a, 'hash'))).toEqual(extractedBuildLogErrors);
+        expect(container.aceEditor.commitState).toBe(CommitState.COULD_NOT_BE_RETRIEVED);
 
         // actions
-        expect(container.actions.commitState).to.equal(CommitState.COULD_NOT_BE_RETRIEVED);
-        expect(container.actions.editorState).to.equal(EditorState.CLEAN);
-        expect(container.actions.isBuilding).to.be.false;
+        expect(container.actions.commitState).toBe(CommitState.COULD_NOT_BE_RETRIEVED);
+        expect(container.actions.editorState).toBe(EditorState.CLEAN);
+        expect(container.actions.isBuilding).toBe(false);
 
         // status
-        expect(container.fileBrowser.status.commitState).to.equal(CommitState.COULD_NOT_BE_RETRIEVED);
-        expect(container.fileBrowser.status.editorState).to.equal(EditorState.CLEAN);
+        expect(container.fileBrowser.status.commitState).toBe(CommitState.COULD_NOT_BE_RETRIEVED);
+        expect(container.fileBrowser.status.editorState).toBe(EditorState.CLEAN);
 
         // build output
-        expect(getBuildLogsStub).to.have.been.calledOnce;
-        expect(container.buildOutput.rawBuildLogs.extractErrors(ProgrammingLanguage.JAVA)).to.deep.equal(extractedBuildLogErrors);
-        expect(container.buildOutput.isBuilding).to.be.false;
+        expect(getBuildLogsStub).toHaveBeenCalledTimes(1);
+        expect(container.buildOutput.rawBuildLogs.extractErrors(ProgrammingLanguage.JAVA)).toIncludeSameMembers(extractedBuildLogErrors);
```

Comment:
```suggestion
        expect(container.buildOutput.rawBuildLogs.extractErrors(ProgrammingLanguage.JAVA)).toEqual(extractedBuildLogErrors);
```
`toEqual` should work here now, too.